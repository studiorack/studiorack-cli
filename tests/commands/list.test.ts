import { expect, test } from 'vitest';
import { cli } from '../shared';
import { RegistryType } from '@open-audio-stack/core';

test('List packages', async () => {
  expect(cli(RegistryType.Plugins, 'list')).toMatchSnapshot();
});
