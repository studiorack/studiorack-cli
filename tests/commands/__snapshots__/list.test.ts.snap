// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`List packages 1`] = `
┌───────────────────────────────────────────┬───────────────────────────┬─────────┬────────────┬───────────────┬───────────────────────────────────┐
│ Id                                        │ Name                      │ Version │ Date       │ License       │ Tags                              │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ wolf-plugins/wolf-spectrum                │ Wolf Spectrum             │ 1.0.0   │ 2019-04-14 │ gpl-3.0       │ Effect, Spectrogram, Frequency    │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ wolf-plugins/wolf-shaper                  │ Wolf Shaper               │ 1.0.2   │ 2023-05-14 │ gpl-3.0       │ Effect, Distortion, Editor        │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ utokusa/os-251                            │ OS-251                    │ 1.2.1   │ 2022-07-25 │ gpl-3.0       │ Instrument, Synth, Lo-fi          │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ tote-bag-labs/valentine                   │ Valentine                 │ 1.0.1   │ 2024-04-27 │ gpl-3.0       │ Effect, Compressor, Distortion    │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ tobanteaudio/modeq                        │ modEQ                     │ 0.4.0   │ 2019-05-17 │ gpl-3.0       │ Effect, EQ, Modulation            │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ thewavewarden/odin                        │ Odin                      │ 2.3.4   │ 2020-11-09 │ gpl-3.0       │ Instrument, Synth, Moog           │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ the-synister/synister                     │ Synister                  │ 1.0.3   │ 2016-03-19 │ gpl-3.0       │ Instrument, Synth, Modulation     │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ tesselode/mika-micro                      │ Mika Micro                │ 2.0.1   │ 2018-05-17 │ mit           │ Instrument, Synth, Oscillators    │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ tesselode/flutterbird                     │ Flutterbird               │ 1.0.1   │ 2018-09-06 │ mit           │ Effect, Pitch, Modulation         │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ tesselode/cocoa-delay                     │ Cocoa Delay               │ 1.0.0   │ 2018-09-01 │ mit           │ Effect, Delay, Warm               │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ surge-synthesizer/surge                   │ Surge XT                  │ 1.3.4   │ 2024-08-11 │ gpl-3.0       │ Instrument, Synth, Modulation     │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ surge-synthesizer/monique-monosynth       │ Monique                   │ 1.2.0   │ 2021-12-13 │ gpl-3.0       │ Instrument, Synth, Bass, Lead     │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ studiorack/avl-percussions                │ AVL Percussions           │ 1.1.0   │ 2022-02-21 │ cc-by-sa-4... │ Instrument, Drums, Percussions... │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ studiorack/avl-drumkits                   │ AVL Drumkits              │ 1.1.0   │ 2022-02-20 │ cc-by-sa-4... │ Instrument, Drums, sfz            │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ sfztools/sfizz                            │ Sfizz                     │ 1.2.3   │ 2024-01-14 │ bsd-2-clau... │ Instrument, Sampler, Synth        │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ sfzinstruments/war-tuba                   │ War Tuba                  │ 1.0.2   │ 2022-11-04 │ cc-by-4.0     │ Instrument, Brass, Tuba, sfz      │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ sfzinstruments/virtuosity_drums           │ Virtuosity Drums          │ 0.9.24  │ 2022-09-21 │ cc0-1.0       │ Instrument, Drums, Jazz, sfz      │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ sfzinstruments/splendidgrandpiano         │ Splendid Grand Piano      │ 1.0.0   │ 2022-01-02 │ bsd-2-clau... │ Instrument, Keys, Piano, sfz      │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ sfzinstruments/samssonor                  │ Sam's Sonor               │ 1.0.0   │ 2022-08-28 │ cc-by-sa-4... │ Instrument, Drums, Kit, sfz       │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ sfzinstruments/salamandergrandpiano       │ Salamander Grand Piano    │ 1.0.0   │ 2022-01-02 │ cc-by-4.0     │ Instrument, Piano, Yamaha, sfz    │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ sfzinstruments/nakeddrums                 │ Naked Drums               │ 1.0.0   │ 2021-11-08 │ cc-by-4.0     │ Instrument, Drums, sfz            │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ sfzinstruments/jsteeldrum                 │ jSteelDrum                │ 1.0.0   │ 2022-09-23 │ unlicense     │ Instrument, Steel, Drum, Orche... │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ sfzinstruments/jrhodes3c                  │ jRhodes3c                 │ 1.0.0   │ 2023-07-04 │ cc-by-sa-4... │ Instrument, Keys, Rhodes, sfz     │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ sfzinstruments/hungarian-zither           │ Hungarian Zither          │ 1.0.1   │ 2022-09-28 │ cc0-1.0       │ Instrument, Strings, Classical... │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ sfzinstruments/horse-pulse                │ Horse Pulse               │ 1.0.0   │ 2022-04-01 │ cc0-1.0       │ Instrument, String, Pluck, sfz    │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ sfzinstruments/double-bass                │ Double Bass               │ 1.0.1   │ 2022-09-04 │ cc0-1.0       │ Instrument, String, Bass, sfz     │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ sfzinstruments/black-and-green-guitars    │ Black and Green Guitars   │ 1.0.0   │ 2022-08-08 │ cc0-1.0       │ Instrument, Guitar, Electric, ... │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ sfzinstruments/bear-sax                   │ Bear Sax                  │ 1.0.4   │ 2021-09-11 │ cc0-1.0       │ Instrument, Brass, Saxophone, ... │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ sdatkinson/neural-amp-modeler             │ Neural Amp Modeler        │ 0.7.12  │ 2024-12-02 │ mit           │ Amp, Emulation, Noise Gate        │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ resonantdsp/swankyamp                     │ Swanky Amp                │ 1.4.0   │ 2021-04-21 │ gpl-3.0       │ Effect, Amplifier, Tube           │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ reales/opl                                │ OPL                       │ 2.4.0   │ 2024-08-08 │ gpl-3.0       │ Instrument, Synth, FM             │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ reales/ob-xd                              │ OB-Xd                     │ 3.5.0   │ 2021-05-30 │ gpl-3.0       │ Instrument, Synth, OB-X           │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ pentagrampro/owlbass                      │ OwlBass                   │ 1.0.0   │ 2019-01-13 │ gpl-3.0       │ Instrument, Synth, Bass           │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ paynebc/tunefish                          │ Tunefish                  │ 4.2.0   │ 2018-05-04 │ gpl-3.0       │ Instrument, Synth, Analog         │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ open-soundfonts/virtual-playing-orchestra │ Virtual Playing Orchestra │ 3.3.2   │ 2020-10-17 │ gpl-3.0       │ Instrument, Orchestral, String... │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ mzuther/squeezer                          │ Squeezer                  │ 2.5.4   │ 2020-04-17 │ gpl-3.0       │ Effect, Compressor                │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ mtytel/helm                               │ Helm                      │ 0.9.0   │ 2017-07-08 │ gpl-3.0       │ Instrument, Synth, Modulation     │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ mikemorenodsp/ep-mk1                      │ EP-MK1                    │ 2.4.0   │ 2020-09-28 │ gpl-3.0       │ Instrument, Synth, Electric, P... │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ midilab/jc303                             │ jc303                     │ 0.11.1  │ 2024-09-11 │ gpl-3.0       │ Instrument, Bass, Synthesizer     │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ michaelwillis/dragonfly-reverb            │ Dragonfly Reverb          │ 3.2.10  │ 2023-04-22 │ gpl-3.0       │ Effect, Reverb, Plate             │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ michael02022/basic-harmonica              │ Basic Harmonica           │ 1.0.0   │ 2022-10-07 │ cc0-1.0       │ Instrument, Wind, Harmonica, O... │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ michael-jan/key-repeat                    │ Key Repeat                │ 1.0.0   │ 2020-01-07 │ gpl-3.0       │ Instrument, Samplers, Repeat      │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ martineastwood/mverb                      │ Mverb                     │ 1.0.0   │ 2013-03-25 │ gpl-3.0       │ Effect, Reverb                    │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ khrykin/blackbird                         │ BlackBird                 │ 0.1.0   │ 2020-11-11 │ gpl-3.0       │ Instrument, Synth, Analog         │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ keijiro/vst2413                           │ VST2413                   │ 1.0.2   │ 2013-04-21 │ zlib          │ Instrument, Synth, Bit            │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ jpcima/opnplug                            │ OPNplug                   │ 1.0.2   │ 2020-08-05 │ bsl-1.0       │ Instrument, Synth, FM             │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ jpcima/adlplug                            │ ADLplug                   │ 1.0.2   │ 2020-08-05 │ bsl-1.0       │ Instrument, Synth, FM             │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ joepvanlier/ysfx                          │ ysfx                      │ 0.0.33  │ 2025-02-09 │ gpl-3.0       │ Instrument, Synth, JSFX           │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ igorski/regrader                          │ Regrader                  │ 1.0.5   │ 2024-03-16 │ mit           │ Effect, Delay, Sync               │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ igorski/fogpad                            │ FogPad                    │ 1.0.3   │ 2024-03-16 │ mit           │ Effect, Pitch Shifter, Reverb     │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ guitarml/smartguitaramp                   │ SmartGuitarAmp            │ 1.3.0   │ 2022-08-31 │ apache-2.0    │ Effect, Guitar, Amp               │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ greatest-ape/octasine                     │ OctaSine                  │ 0.9.1   │ 2024-07-03 │ agpl-3.0      │ Instrument, Synth, Modulation     │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ freepats/upright-piano-kw                 │ Upright Piano KW          │ 1.0.0   │ 2022-02-21 │ cc0-1.0       │ Instrument, Keys, Piano, sfz      │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ freepats/synthesizer-percussion           │ Synthesizer Percussion    │ 1.0.0   │ 2022-07-18 │ cc0-1.0       │ Instrument, Drums, Electro, sf... │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ freepats/spanish-classical-guitar         │ Spanish Classical Guitar  │ 1.0.0   │ 2019-06-18 │ cc0-1.0       │ Instrument, Guitar, Classical,... │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ freepats/hang-d-minor                     │ Hang D Minor              │ 1.0.0   │ 2022-03-30 │ cc0-1.0       │ Instrument, Orchestra, Hang, s... │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ freepats/glasses-of-water                 │ Glasses Of Water          │ 1.0.0   │ 2019-12-27 │ cc0-1.0       │ Instrument, Orchestra, Glass, ... │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ falktx/carla                              │ Carla                     │ 2.5.9   │ 2024-09-22 │ gpl-3.0       │ Host, Player, Automation          │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ endolith/salamander-drumkit               │ Salamander Drumkit        │ 1.0.0   │ 2012-02-25 │ cc-by-sa-4... │ Instrument, Drums, sfz            │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ dropsnorz/wobbleizer                      │ Wobbleizer                │ 2.3.2   │ 2012-02-25 │ mit           │ Effect, Filter, LFO               │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ distrho/dpf-plugins                       │ DPF Plugins               │ 1.7.0   │ 2023-04-15 │ gpl-3.0       │ Effect, Delay, Feedback, Rever... │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ clearly-broken-software/ninjas2           │ Ninjas 2                  │ 0.2.0   │ 2020-01-17 │ gpl-3.0       │ Instrument, Sampler, Slicer       │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ clearly-broken-software/drops             │ Drops                     │ 1.0.0   │ 2022-01-15 │ gpl-3.0       │ Instrument, Sampler, Filter       │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ chowdhury-dsp/chowtapemodel               │ Chow Tape Model           │ 2.11.4  │ 2023-11-05 │ gpl-3.0       │ Effect, Filter, Tape              │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ chowdhury-dsp/chowmatrix                  │ CHOW Matrix               │ 1.3.0   │ 2021-12-21 │ bsd-3-clau... │ Effect, Delay, Feedback           │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ chowdhury-dsp/chow                        │ CHOW                      │ 1.0.0   │ 2019-01-11 │ gpl-3.0       │ Effect, Distortion, Noise         │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ birch-san/juicysfplugin                   │ JuicySFplugin             │ 3.1.0   │ 2022-02-13 │ gpl-3.0       │ Instrument, Sampler, Synth        │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ audioplugins/iempluginsuite               │ IEM Plugin Suite          │ 1.14.1  │ 2023-02-23 │ gpl-3.0       │ Effect, Reverb, Delay             │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ asb2m10/dexed                             │ Dexed                     │ 0.9.8   │ 2024-10-08 │ gpl-3.0       │ Instrument, Synth, FM             │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ artfwo/andes                              │ Andes                     │ 0.2.0   │ 2018-06-01 │ gpl-3.0       │ Instrument, Synth, Oscillator     │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ ardura/subhoofer                          │ Subhoofer                 │ 2.2.2   │ 2024-08-02 │ gpl-3.0       │ Effect, Distortion                │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ ardura/scrollscope                        │ Scrollscope               │ 1.4.1   │ 2024-07-12 │ gpl-3.0       │ Effect, Analysis                  │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ ardura/interleaf                          │ Interleaf                 │ 1.0.0   │ 2024-07-12 │ gpl-3.0       │ Effect, EQ                        │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ ardura/gladedesk                          │ Glade Desk                │ 1.0.0   │ 2024-07-19 │ gpl-3.0       │ Effect, Console, Emulation        │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ ardura/actuate                            │ Actuate                   │ 1.3.5   │ 2024-07-12 │ gpl-3.0       │ Instrument, Synth, Sampler        │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ antonok-edm/ampli-fe                      │ ampli-Fe                  │ 0.1.1   │ 2021-11-02 │ mit           │ Effect, Amplifier, Volume         │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ airwindows/airwindows                     │ Airwindows                │ 1.0.0   │ 2020-02-20 │ mit           │ Effect, Chorus, Distortion, EQ... │
├───────────────────────────────────────────┼───────────────────────────┼─────────┼────────────┼───────────────┼───────────────────────────────────┤
│ aaronaanderson/terrain                    │ Terrain                   │ 1.2.2   │ 2024-11-01 │ gpl-3.0       │ Instrument, Synthesizer, Wavet... │
└───────────────────────────────────────────┴───────────────────────────┴─────────┴────────────┴───────────────┴───────────────────────────────────┘
`;

exports[`List packages 2`] = `No results found`;
